packages/e2e-tests/config/setup-screenshots.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const root = process.env.GITHUB_WORKSPACE || '../../../';
+const SCREENSHOTS_PATH = root + '/screenshots';
+
+if ( ! fs.existsSync( SCREENSHOTS_PATH ) ) {
+	fs.mkdirSync( SCREENSHOTS_PATH );
+}
+
+/**
+ * @copyright Tom Esterez (@testerez) https://github.com/smooth-code/jest-puppeteer/issues/131#issuecomment-424073620
+ */
+export const registerScreenshotReporter = () => {
+	/**
+	 * jasmine reporter does not support async.
+	 * So we store the screenshot promise and wait for it before each test
+	 */
+	let screenshotPromise = Promise.resolve();
+	beforeEach( () => screenshotPromise );
+	afterAll( () => screenshotPromise );
+
+	/**
+	 * Take a screenshot on Failed test.
+	 * Jest standard reporters run in a separate process so they don't have
+	 * access to the page instance. Using jasmine reporter allows us to
+	 * have access to the test result, test name and page instance at the same time.
+	 */
+	jasmine.getEnv().addReporter( {
+		specDone: async ( result ) => {
```

Comment:
this probably doesn't have to be `async` anymore since we're dealing with raw promises